### PR TITLE
Add bgutil copy <src> <dst> (#263)

### PR DESCRIPTION
**Pull request by Thib17**
This change is [Reviewable](https://reviewable.io/reviews/criteo/biggraphite/266)

### CODE REVIEW

**Comment by iksaif** [comment]
Add small unit test please :) (with the fakeaccessor)

**Comment by iksaif** [comment]
Also travis is failing

**Comment by Thib17** [comment]
unit test added

---

Review status: 0 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/criteo/biggraphite/266#-:-KhX01TViNDPD1t3d0dm:bxnzd61)*

File: biggraphite/cli/command_copy.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            default=datetime.datetime.now(),
+            required=False,
+        )
+        parser.add_argument(
+            "--dry-run",
+            action="store_const", default=False, const=True,
+            help="Only show commands to create/upgrade the schema.",
+            required=False,
+        )
+        parser.add_argument(
```
Comment (iksaif) [inline biggraphite/cli/command_copy.py]: We already have a flag to do that (at the bgutil level)

File: biggraphite/cli/command_copy.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        time_start = time.mktime(opts.time_start.timetuple())
+        time_end = time.mktime(opts.time_end.timetuple())
+
+        for src_metric, dst_metric in metric_tuples:
+            self._copy_metric(accessor, src_metric, dst_metric, time_start, time_end)
+
+    @staticmethod
+    def _get_metric_tuples(accessor, src, dst, recursive, dry_run=True):
+        pattern = "%s.**" % src if recursive else src
```
Comment (iksaif) [inline biggraphite/cli/command_copy.py]: There is a flag, so that's good, forget this comment.

File: biggraphite/cli/command_copy.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        try:
+            src_metrics = list_metrics(accessor, pattern)
```
Comment (iksaif) [inline biggraphite/cli/command_copy.py]: list_metrics() returns a list, not a generator. If you want to make it work with large numbers of metrics you'll need to change it to return a generator

File: tests/test_command_copy.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        cmd_copy = command_copy.CommandCopy()
+
+        # Chack that _METRIC_2 is empty
+        for i in range(3):
+            pts = self.accessor.fetch_points(
+                self._METRIC_2,
+                self._POINTS_START, self._POINTS_END,
+                stage=self._METRIC_2.retention[i]
+            )
+            self.assertEqual([p for p in pts], [])
```
Comment (iksaif) [inline tests/test_command_copy.py]: you can just do list(pts)

File: biggraphite/cli/command_copy.py (hunk 3 above)
Comment (Thib17) [inline biggraphite/cli/command_copy.py]: done

File: tests/test_command_copy.py (hunk 4 above)
Comment (Thib17) [inline tests/test_command_copy.py]: done

**Pull request merged**